job-service-contract/src/main/resources/com/hpe/caf/services/job/swagger.yaml

```diff
@@ -335,6 +337,55 @@ definitions:
         example:
           "tag:4": "4"
           owner: "bob"
+  expirationPolicy:
+    type: object
+    properties:
+      "Active":
+        $ref: "#/definitions/policy"
+      "Completed":
+        $ref: "#/definitions/policy"
+      "Failed":
+        $ref: "#/definitions/policy"
+      "Cancelled":
+        $ref: "#/definitions/policy"
+      "Waiting":
+        $ref: "#/definitions/policy"
+      "Paused":
+        $ref: "#/definitions/policy"
+      "Expired":
+        $ref: "#/definitions/deletePolicy"
+      "Default":
+        $ref: "#/definitions/policy"
+    description: The expiration policy to be applied on the job
+  deletePolicy:
+    type: object
+    properties:
+      expiryTime:
+        type: string
+        pattern: '^((lastUpdateDate)|(createDate))(\+P)([1-9]\d?|[12]\d{2}|3[0-5]\d|36[0-5])D$'
+        default: "createDate+10D"
+      expirationOperation:
+        type: string
+        enum:
+          - delete
+    description: The expiration details to be applied on the job
+  policy:
+    type: object
+    properties:
+      expiryTime:
+        type: string
+        pattern: '^((lastUpdateDate)|(createDate))(\+P)([1-9]\d?|[12]\d{2}|3[0-5]\d|36[0-5])D$'
```

Comment:
- The contract uses [`lastUpdateTime`](https://github.com/JobService/job-service/blob/v3.5.0-source/job-service-contract/src/main/resources/com/hpe/caf/services/job/swagger.yaml#L370) and [`createTime`](https://github.com/JobService/job-service/blob/v3.5.0-source/job-service-contract/src/main/resources/com/hpe/caf/services/job/swagger.yaml#L366).  It makes no sense to use `lastUpdateDate` and `createDate` as they are not the property names presented to the consumer.
- I'm not going to study this regular-expression but I can see that it ends in `D`, so I assume it only supports days.  I thought we were going to use the RFC 3339 duration format.  I should be able to specify `lastUpdateTime+P20M` to expire after 20 minutes but this regular expression is clearly not going to accept that.
- What happened with supporting an exact time?  In [this](https://github.com/JobService/job-service/pull/235#discussion_r590916801) comment I said: "It could be an exact time like `2021-04-12T23:20:50.52Z`, or it could be an expression like `lastUpdateTime+P1D`".  In [this](https://github.com/JobService/job-service/pull/235#discussion_r592019221) one I said: "I could imagine that we might want to pass an exact expiry time...".  These comments appear to have been ignored and you're now trying to lock us in to a duration instead of supporting both like I repeatedly said.
- Finally, just for the record I can see the attractiveness of specifying a pattern if you can get it right, but the reality is that it is a nice-to-have.  I'd be perfectly happy with a few decent examples in the description.

Reply:
- Updated for  lastUpdateTime and createTime.
- Adapted regex for accepting RFC3339 format date or lastUpdateTime+P1[DYHM] pattern (https://regex101.com/r/badcKz/1/)
- Added a few examples

Reply:
The regex may still need some adjustment - I notice that it still matched when I added "xx" at the start:

![image](https://user-images.githubusercontent.com/20627297/110981497-2329b600-835f-11eb-8d83-781567570026.png)